service/files/files_api.go

```diff
@@ -0,0 +1,80 @@
+package files
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/databricks/databricks-sdk-go/client"
+)
+
+// BEGIN TODO: remove this once FilesAPI gets OpenAPI spec
+
+func NewFiles(client *client.DatabricksClient) *FilesAPI {
+	return &FilesAPI{
+		impl: &filesImpl{
+			client: client,
+		},
+	}
+}
+
+type FilesService interface{}
+
+type FilesAPI struct {
+	impl FilesService
+}
+
+type filesImpl struct {
+	client *client.DatabricksClient
+}
+
+// END TODO: remove this once FilesAPI gets OpenAPI spec
+
+func (a *FilesAPI) Upload(ctx context.Context, path string, r io.Reader) error {
+	impl, ok := a.impl.(*filesImpl)
+	if !ok {
+		return fmt.Errorf("wrong impl: %v", a.impl)
+	}
+	return impl.client.Do(ctx, "PUT", "/api/2.0/fs/files"+path, r, nil,
+		func(r *http.Request) error {
+			r.Header.Set("Content-Type", "application/octet-stream")
+			return nil
+		})
+}
+
+// WriteFile is identical to [os.WriteFile] but for Files API.
+func (a *FilesAPI) WriteFile(ctx context.Context, name string, data []byte) error {
+	return a.Upload(ctx, name, bytes.NewBuffer(data))
+}
+
+func (a *FilesAPI) Download(ctx context.Context, path string) (*bytes.Buffer, error) {
+	impl, ok := a.impl.(*filesImpl)
+	if !ok {
+		return nil, fmt.Errorf("wrong impl: %v", a.impl)
+	}
+	var buf bytes.Buffer
+	err := impl.client.Do(ctx, "GET", "/api/2.0/fs/files"+path, nil, &buf)
```

Comment:
This still requires buffering of the whole response, which can be very large.

Especially if we have the `ReadFile` function we can make `Download` return an `io.Reader` and allow the caller to stream the response.

Reply:
we'll fix it if it becomes a problem

Reply:
Ack, works for me.